src/plugins/local/selectors/localSelectors.js

```diff
@@ -48,18 +85,21 @@ export const filteredDataSelector = createSelector(
       return data;
     }
 
-    const filterToLower = filter.toLowerCase();
-    return data.filter(row =>
-      row.keySeq()
-        .some((key) => {
-          const filterable = columnProperties && columnProperties.getIn([key, 'filterable']);
-          if (filterable === false) {
-            return false;
-          }
-          const value = row.get(key);
-          return value &&
-            value.toString().toLowerCase().indexOf(filterToLower) > -1;
-        }));
+    return data.filter(row => {
+      switch (typeof (filter)) {
```

Comment:
For performance, can you flip this to process `filter` outside the `data.filter(...)`? This would work particularly well if you curry the functions above, e.g.

```javascript
switch (typeof filter) {
  case 'string':
    return data.filter(textFilterRowSearch(columnProperties, filter));
  case 'object':
    return data.filter(objectFilterRowSearch(columnProperties, filter));
  case 'function':
    return data.filter(filter);
  default:
    return data;
}
```

Reply:
Will get onto it!